custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/src/main/java/com/datarobot/drum/PredictorEntryPoint.java

```diff
@@ -86,15 +86,18 @@ private static EntryPointConfig parseArgs(String[] args) throws Exception {
     }
 
     public static void main(String[] args) throws Exception {
+        int portNumber = -1;
         try {
             EntryPointConfig config = parseArgs(args);
+            portNumber = config.portNumber;
             PredictorEntryPoint entryPoint = new PredictorEntryPoint(config.className, config.predictorName);
             // TODO: use a specific port, note multiple such gateways might be running.
             GatewayServer gatewayServer = new GatewayServer(entryPoint, config.portNumber);
             gatewayServer.start();
         } catch (Exception e) {
-            System.out.println("Got exception in ComponentEntryPoint: " + e.getMessage());
-            throw e;
+            System.out.println(String.format("PredictorEntryPoint failed to start py4j GatewayServer on port: %d", portNumber));
```

Comment:
Don't you want to print the message from the exception? (.e.g `e.getMessage()`)

Reply:
In the example I play with, trying to bind to port 80.
```
null
PredictorEntryPoint failed to start py4j GatewayServer on port: 80
py4j.Py4JNetworkException
	at py4j.GatewayServer.startSocket(GatewayServer.java:788)
	at py4j.GatewayServer.start(GatewayServer.java:763)
	at py4j.GatewayServer.start(GatewayServer.java:746)
	at com.datarobot.drum.PredictorEntryPoint.main(PredictorEntryPoint.java:96)
Caused by: java.net.BindException: Permission denied (Bind failed)
	at java.base/java.net.PlainSocketImpl.socketBind(Native Method)
	at java.base/java.net.AbstractPlainSocketImpl.bind(AbstractPlainSocketImpl.java:436)
	at java.base/java.net.ServerSocket.bind(ServerSocket.java:395)
	at py4j.GatewayServer.startSocket(GatewayServer.java:786)
	... 3 more

```

In this exception message passed is `null`, that's how it is implemented:

https://github.com/bartdag/py4j/blob/366de7a527df122f4c5b50a5aa10fb69cd47a089/py4j-java/src/main/java/py4j/GatewayServer.java#L788

https://github.com/bartdag/py4j/blob/366de7a527df122f4c5b50a5aa10fb69cd47a089/py4j-java/src/main/java/py4j/Py4JNetworkException.java#L79

Reply:
Or do you mean to print the message in the addition to the stacktrace?

Reply:
added message printing